fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MaterializationContext.java

```diff
@@ -370,8 +371,16 @@ public int compare(MaterializationContext o1, MaterializationContext o2) {
             OperatorType o2Type = o2.getMvExpression().getOp().getOpType();
 
             if (o1Type == o2Type && (o1Type == OperatorType.LOGICAL_AGGR)) {
+                boolean mvHasDifferentRows = orderingRowCount(o1) != 0 && orderingRowCount(o2) != 0
+                        && orderingRowCount(o1) != orderingRowCount(o2);
                 return Comparator
-                        .comparing(this::orderingAggregation)
+                        .comparing((Function<MaterializationContext, Long>) mv -> {
+                            int r = orderingAggregation(mv);
+                            if (r > 0 && mvHasDifferentRows) {
+                                return orderingRowCount(mv);
+                            }
+                            return (long) r;
+                        })
```

Comment:
eg:
```
-- rows: 100
mv1:  select sum(v1) from t1 group by a, b, c;
-- rows: 10000
mv2: select sum(v1) from t1 group by a, b;

query: select sum(v1) from t1 where b = 'a' group by a;
```

Before: prefer mv with less dimensions -> mv2, but mv2 has more rows which is not expected.
After: when many mvs satisfy query, prefer mv with less rows, so choose `mv1`.

am I right? can you add more comments about this?

Reply:
And can you move this codes into `orderingAggregation`  to make it more clear?

Reply:
Yes，this is what I expect

Reply:
`orderingAggregation` is about single mv, while these codes are about 2 mvs, so I don't change logic in `orderingAggregation`

Reply:
Seems solid.
Can you add this example into comments?